nixos/modules/services/web-servers/nginx/default.nix

```diff
@@ -859,7 +859,7 @@ in
         PrivateMounts = true;
         # System Call Filtering
         SystemCallArchitectures = "native";
-        SystemCallFilter = "~@cpu-emulation @debug @keyring @ipc @mount @obsolete @privileged @setuid";
+        SystemCallFilter = [ "~@cpu-emulation @debug @keyring @ipc @mount @obsolete @privileged @setuid" "pipe" "pipe2" ];
```

Comment:
Is this list not blacklisting syscalls (leading ~)? You would actually blacklist the pipe syscall here.

Reply:
Correct.

Reply:
Result:
```
# Blocked filters
SystemCallFilter=~@cpu-emulation @debug @keyring @ipc @mount @obsolete @privileged @setuid
# Allowed filters
SystemCallFilter=pipe
SystemCallFilter=pipe2

```

Reply:
Oh right, I always miss strings ending and starting in these diffs. If you wanted to make it clearer, add newlines.

But I still don't think this change is valid.

Reply:
Yes. I also think that the pipe syscall is already allowed since its not explicitly blocked.

Reply:
It is blocked as part of the deny list that includes `@ipc`.

Reply:
Current active filters:
```
systemctl show nginx | grep SystemCallFilter
SystemCallFilter=~_sysctl acct add_key adjtimex afs_syscall bdflush bpf break capset chown chown32 chroot clock_adjtime clock_adjtime64 clock_settime clock_settime64 create_module delete_module fanotify_init fanotify_mark fchown fchown32 fchownat finit_module fsconfig fsmount fsopen fspick ftime get_kernel_syms getpmsg gtty idle init_module ioperm iopl ipc kexec_file_load kexec_load keyctl lchown lchown32 lock lookup_dcookie memfd_create modify_ldt mount move_mount mpx mq_getsetattr mq_notify mq_open mq_timedreceive mq_timedreceive_time64 mq_timedsend mq_timedsend_time64 mq_unlink msgctl msgget msgrcv msgsnd nfsservctl open_by_handle_at open_tree pciconfig_iobase pciconfig_read pciconfig_write perf_event_open pidfd_getfd pipe pipe2 pivot_root process_vm_readv process_vm_writev prof profil ptrace putpmsg query_module quotactl reboot request_key rtas security semctl semget semop semtimedop semtimedop_time64 setdomainname setfsuid setfsuid32 setgid setgid32 setgroups setgroups32 sethostname setregid setregid32 setresgid setresgid32 setresuid setresuid32 setreuid setreuid32 settimeofday setuid setuid32 sgetmask shmat shmctl shmdt shmget ssetmask stime stty subpage_prot swapoff swapon sys_debug_setcontext sysfs tuxcall ulimit umount umount2 uselib ustat vhangup vm86 vm86old vserver
```
`pipe` and `pipe2` filtered.

Reply:
Can we please discuss the points I raised in https://github.com/NixOS/nixpkgs/pull/123649#issuecomment-843941451.